### ISSUE TITLE
Shouldn't this need hooks?

### ISSUE THREAD

**Issue by prateekbh** [preactjs/preact-router#339]
Should we add hooks in this?

some hooks like `useRouter` would make sense.

```js
const [location, setLocation] = useRouter();
```

This would allow access of current location on all components and mimic current `route` functionality by setLocation.

**Comment by marvinhagemeister** [comment]
I really like the API [wouter](https://github.com/molefrog/wouter) has chosen. They do ship with Preact support too :+1:

**Comment by developit** [comment]
Yup. I'm not sure it makes sense to completely change the API of this module when there's an existing implementation we like.

**Comment by gtsop** [comment]
So what is the suggestion here? We use both preact-router and wouter or just wouter?

**Comment by marvinhagemeister** [comment]
@gtsop either router is a great choice 👍 Just don't try to use them at the same time, that always leads to problems regardless of the framework you're using.

**Comment by developit** [comment]
I think it might be worth exporting hooks from `preact-router/match`. It's already an add-on lib, and the "hooks" are already there to be able to implement reactivity. It likely wouldn't even need to use the new context API.

**Comment by toniopelo** [comment]
> I think it might be worth exporting hooks from `preact-router/match`. It's already an add-on lib, and the subscription mechanism is already there to be able to implement reactivity. It likely wouldn't even need to use the new context API.

That would be great @developit 
I'm new to preact and maybe I'm not enough familiar with it yet. But I feel like this simple use case is not covered:

``` 
<Component class={currentPath === '/some/path' ? 'active' : ''}>
    ...
</Component>
```

In some situation `Component` cannot be changed with `Link` because it's not a link and `<Match></Match>` can't do the job without duplicating code (one `Component` with 'active' and another `Component` without 'active'). And probably even causing component recreation when my snippet wouldn't ? (not sure on this one)

Anyway it feels like having access to the match internals through the use of a hook would be great 👍 
Would be happy to submit a PR if I find the time for it ⏳ 🙈

**Comment by toniopelo** [comment]
> I think it might be worth exporting hooks from `preact-router/match`. It's already an add-on lib, and the subscription mechanism is already there to be able to implement reactivity. It likely wouldn't even need to use the new context API.

Created a PR for this but I finally decided to use the Context API. The reason is to be able to keep this hook working when your app uses several Router instances. With the Context API I can ensure that I get the values corresponding to the router I'm depending on.

The PR: #370
